packages/lodestar/src/chain/chain.ts

```diff
@@ -140,7 +140,7 @@ export class BeaconChain extends (EventEmitter as { new(): ChainEventEmitter })
     );
     // Determine whether a genesis state already in
     // the database matches what we were provided
-    const storedGenesisBlock = await this.db.block.getBlockBySlot(GENESIS_SLOT);
+    const storedGenesisBlock = await this.db.block.getBySlot(GENESIS_SLOT);
```

Comment:
This could be in blockArchive if we have finalized chain stored

Reply:
agreed, the genesis handling is not right, but trying to limit the scope of this PR to just fix up the db controller and not change too much outside that.